### PR TITLE
encryption: fix issue with opening plaintext files

### LINKED ISSUE #12162: Failed to create raft engine when encryption enabled.
### What version of TiKV are you using?

```
Release Version:   6.0.0-alpha
Edition:           Community
Git Commit Hash:   1047b51285456e702b66253da26c7f14d582ff0c
Git Commit Branch: master
UTC Build Time:    2022-03-14 08:48:38
Rust Version:      rustc 1.60.0-nightly (1e12aef3f 2022-02-13)
Enable Features:   jemalloc mem-profiling portable sse test-engines-rocksdb cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
```

### Steps to reproduce

add some config in tikv.toml and then starts tikv-server
```
[security.encryption]
data-encryption-method = "aes256-ctr"

[security.encryption.master-key]
type = "file"
path = "/tmp/backup_restore_test/master-key-file"
```
### What did you expect?
tikv serves well.

### What did happened?
```
[2022/03/15 12:37:28.488 +08:00] [FATAL] [[server.rs:1372](http://server.rs:1372/)] ["Failed to create raft engine: \"[components/raft_log_engine/src/[engine.rs:440](http://engine.rs:440/)]: IO Error: Other error [components/encryption/src/[crypter.rs:83](http://crypter.rs:83/)]: Nonce + Counter must be 12/16 bytes, 0\""]
```

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

### What is changed and how it works?

Issue Number: Close #12162

What's Changed:

When key manager doesn't have the key for certain data files, we should still be able to open it.

```commit-message
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/encryption/src/manager/mod.rs [hunk 1]
```diff
@@ -591,7 +593,12 @@ impl DataKeyManager {
             writer,
             crypter::encryption_method_from_db_encryption_method(file.method),
             &file.key,
-            Iv::from_slice(&file.iv)?,
+            if file.iv.is_empty() {
```
Comment (glorv) [inline components/encryption/src/manager/mod.rs:596]: I'm concerned about adding this logic will hide bugs that forget to register file name and thus the target file will be not encrypted when encryption is enabled. Are there any case we actually need this fallback logic?

Comment (tabokie) [reply components/encryption/src/manager/mod.rs:596]: That's a fair point. But right now RocksDB also allows opening plaintext file:
https://github.com/tikv/rocksdb/blob/21712fe4243a57a5e7854574b933e67d0fd227f6/encryption/encryption.cc#L262. The two should be consistent.

With this patch, we should be able to enable encryption on an un-encrypted dataset. I honestly don't like using encryption this way. But QA thinks it's useful for their testing.

**Comment by tabokie** [comment]
Not needed now.

**Comment by tabokie** [comment]
Ping @glorv @Connor1996

**Review by glorv — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**